docs/express-or-core.qmd

```diff
@@ -0,0 +1,267 @@
+
+At this time, Shiny Core's functionality is a superset of Express, meaning that anything you can do in Express, you can also do in Core.
+The reverse is not true, however.
+
+Most importantly, [Shiny Modules](modules.qmd) are supported in Shiny Core but not (yet) in Shiny Express. Shiny Modules are extremely useful for organizing large apps into smaller, more manageable pieces, and are also a mechanism for reusing Shiny application logic.
+
+Shiny Core also has `ui.insert_ui()` and `ui.remove_ui()` functions, which is a way to imperatively add or remove UI elements from the app at any time.
+Despite being available in `shiny.express.ui`, these functions do not currently work well with Shiny Express. The same goes for `ui.modal_show()`.
+
+## Maturity
+
+**Consider using Shiny Core if you care more about maturity and stability than convenience.**
+
+Given its longer history, Shiny Core is naturally more mature than Shiny Express in both syntax and implementation.
+
+We've carefully designed the Shiny Express syntax, and hope not to have to make breaking changes to it.
+However, we don't know what we don't know, and it's possible that user feedback or our own testing will someday require us to make significant changes.
+
+Similarly, we are constantly testing Shiny Express, but as of this writing, it has not has as much real-world use as Shiny Core.
+Therefore, with Shiny Core, you are less likely to encounter bugs.
+
+## Familiarity to R users
+
+**Consider using Shiny Core if you are an R user who is already familiar with Shiny.**
+
+While Shiny Core is not a literal translation of Shiny for R, it is much closer to it than Shiny Express.
+The UI/server separation, the nested UI function calls, the matching of output IDs to render functions, are all going to feel very natural to experienced Shiny for R app authors.
+
+<!--
+The [next article](express-in-depth.qmd) illustrates how these downsides can be mitigated to some extent in Express with some advanced tooling.
+However, if you feel yourself reaching for these tools often, it might be time to [switch to Core](express-to-core.qmd).
+-->
+
+## Appendix
+
+The following is the dashboard application from the Essentials section of this guide, rewritten using Shiny Core.
+Compare it to [the original](user-interfaces.qmd#all-together-now).
+
+::: {.column-page-right}
+
+```{shinylive-python}
+#| standalone: true
+#| components: [editor, viewer]
+#| layout: vertical
+#| viewerHeight: 800
+import faicons as fa
+import plotly.express as px
+from shinywidgets import output_widget, render_plotly
+
+from shiny import App, reactive, render, req, ui
+
+# Load data and compute static values
+tips = px.data.tips()
+bill_rng = (min(tips.total_bill), max(tips.total_bill))
+
+ICONS = {
+    "user": fa.icon_svg("user", "regular"),
+    "wallet": fa.icon_svg("wallet"),
+    "currency-dollar": fa.icon_svg("dollar-sign"),
+    "gear": fa.icon_svg("gear")
+}
+
+app_ui = ui.page_sidebar(
```

Comment:
Do we want to break this UI object into variables or use a function to generate the UI? It might be nice to demonstrate that functionality.

Reply:
I didn't want to have to introduce that here, and definitely didn't want to _not_ introduce it and then use it.

Incidentally, I personally prefer it this way, one big expression unless the depth gets out of hand or there's something specific you want to isolate conceptually.